cmd/system-probe/main.go

```diff
@@ -85,6 +87,14 @@ func runAgent(exit <-chan struct{}) {
 		gracefulExit()
 	}
 
+	if cfg.ProfilingEnabled {
```

Comment:
How do you guys want to deal with the config setting for our customers? With the core agent we decided to include it on the config template as opposed to keeping it as a "secret" option because it incurs in a cost both from a monetary and performance aspect: https://github.com/DataDog/datadog-agent/pull/5476/files#diff-a77843a3a2d99bdc21000210a1c6a0fc99871774b33d15b6779389e8299aa794R549

We didn't want a customer enabling it by mistake or because they see a config snippet online or whatever and then get angry about having to pay. We were also explicit about it when customers enabled it at runtime: https://github.com/DataDog/datadog-agent/pull/5476/files#diff-a7b073db0b85b1d5cb92c08d38299b1deda96dd38c1bb1b06c98b3bf87179efbR170. 

I'm not sure if you guys want to do something like that, or if you want to use the same setting to decide if you want to enable profiling. Should we use a shared setting for all agents, or should we have per-agent granularity? (I guess the latter makes sense, but then we should probably document in the config template it so that we're consistent).

Reply:
We decided against using the main agent config file settings for a couple reasons:

1. We don't currently read the main agent config file in the system-probe. It would be a more complicated change, and would require changes to helm charts to even expose the config file to system-probe.
1. Since profiling does incur costs, we wanted the system-probe to be explicitly targeted.

Reply:
This was mostly intended for internal use only, but I agree that accidental costs for a customer are not great.

@leeavital Do you have thoughts here?

Reply:
Including it in the system-probe template with a disclaimer about it incurring costs makes sense and is as consistent as possible with the core agent's approach.

Eventually the core agent will need to make runtime settings available to the system-probe in which case we should use that, but that's a ways off.

Reply:
I added the section, but I noticed it isn't included by default for the core agent.